validation-test/compiler_crashers_2_fixed/parser-cutoff.swift

```diff
@@ -0,0 +1,791 @@
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+if a > 1 {
+
+// RUN: not %swift-frontend -typecheck %s
```

Comment:
The crash here was the assertion failure right? Would it be worth adding `// REQUIRES: asserts` or a comment to that effect in here? The test does work either way so maybe doesn't matter, but IMO it'd be nice to differentiate it from the existing `structure_overflow*.swift` ones.

Reply:
Added a comment.